src/main/java/formflow/library/data/SubmissionRepositoryService.java

```diff
@@ -17,67 +19,125 @@
-  }
-
-  /**
-   * Searches for a particular Submission by its {@code id}
-   *
-   * @param id id of submission to look for, not null
-   * @return Optional containing Submission if found, else empty
-   */
-  public Optional<Submission> findById(UUID id) {
-    Optional<Submission> submission = repository.findById(id);
-    return submission.map(value -> encryptionService.decrypt(value));
-  }
-
-  /**
-   * Removes the CSRF from the Submission's input data, if found.
-   *
-   * @param submission submission to remove the CSRF from, not null
-   */
-  public void removeFlowCSRF(Submission submission) {
-    submission.getInputData().remove("_csrf");
-  }
-
-  /**
-   * Removes the CSRF from a particular Submission's Subflow's iteration data, if found.
-   * <p>
-   * This will remove the CSRF from all the iterations in the subflow.
-   * </p>
-   *
-   * @param submission  submission to look for subflows in, not null
-   * @param subflowName the subflow to remove the CSRF from, not null
-   */
-  public void removeSubflowCSRF(Submission submission, String subflowName) {
-    var subflowArr = (ArrayList<Map<String, Object>>) submission.getInputData().get(subflowName);
-
-    if (subflowArr != null) {
-      for (var entry : subflowArr) {
-        entry.remove("_csrf");
-      }
+
+    /**
+     * Searches for a particular Submission by its {@code id}
+     *
+     * @param id id of submission to look for, not null
+     * @return Optional containing Submission if found, else empty
+     */
+    public Optional<Submission> findById(UUID id) {
+        Optional<Submission> submission = repository.findById(id);
+        return submission.map(value -> encryptionService.decrypt(value));
+    }
+
+    public Optional<Submission> findByShortCode(String shortCode) {
+        Optional<Submission> submission = repository.findSubmissionByShortCode(shortCode);
+        return submission.map(value -> encryptionService.decrypt(value));
+    }
+
+    /**
+     * Removes the CSRF from the Submission's input data, if found.
+     *
+     * @param submission submission to remove the CSRF from, not null
+     */
+    public void removeFlowCSRF(Submission submission) {
+        submission.getInputData().remove("_csrf");
+    }
+
+    /**
+     * Removes the CSRF from a particular Submission's Subflow's iteration data, if found.
+     * <p>
+     * This will remove the CSRF from all the iterations in the subflow.
+     * </p>
+     *
+     * @param submission  submission to look for subflows in, not null
+     * @param subflowName the subflow to remove the CSRF from, not null
+     */
+    public void removeSubflowCSRF(Submission submission, String subflowName) {
+        var subflowArr = (ArrayList<Map<String, Object>>) submission.getInputData().get(subflowName);
+
+        if (subflowArr != null) {
+            for (var entry : subflowArr) {
+                entry.remove("_csrf");
+            }
+        }
+    }
+
+    /**
```

Comment:
There was a bunch of reformatting above, but this is where the new method begins.